publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationFromImportCandidateHandler.java

```diff
@@ -101,148 +94,67 @@ protected Integer getSuccessStatusCode(ImportCandidate input, PublicationRespons
-    }
-
-    private boolean changeInCristinIdentifier(Contributor contributor,
-                                              List<Contributor> rawImportCandidateContributors) {
-        return rawImportCandidateContributors.stream()
-                   .anyMatch(rawContributor -> hasSameAuidButDifferentCristinId(rawContributor,
-                                                                                contributor));
-    }
-
-    private boolean hasSameAuidButDifferentCristinId(Contributor rawContributor, Contributor contributor) {
-        return hasDifferentCristinId(rawContributor, contributor)
-               && hasSameAuid(rawContributor, contributor);
-    }
-
-    private boolean hasSameAuid(Contributor rawContributor, Contributor contributor) {
-        var userAuid = extractAuid(contributor);
-        var rawAuid = extractAuid(rawContributor);
-        return Objects.equals(userAuid, rawAuid);
-    }
-
-    private AdditionalIdentifierBase extractAuid(Contributor contributor) {
-        return contributor.getIdentity().getAdditionalIdentifiers().stream().filter(
-            additionalIdentifier -> "scopus-auid".equals(additionalIdentifier.sourceName())
-        ).findFirst().orElse(null);
-    }
-
-    private void copyArtifacts(Publication publication, ImportCandidate importCandidate) {
-        importCandidate.getAssociatedArtifacts().stream()
-            .filter(File.class::isInstance)
-            .filter(file -> wasKeptByImporter((File) file, publication))
-            .map(File.class::cast)
-            .forEach(
-                a -> copyS3file(importCandidateStorageBucket, persistedStorageBucket, a.getIdentifier().toString()));
-    }
+        var resourceToImport = ImportCandidateEnricher.createResourceToImport(requestInfo, importCandidate,
+                                                                              rawImportCandidate);
+
+        if (!resourceToImport.getFiles().isEmpty()) {
+            var customerRequiringApproval = customerRequiringFileApproval(importCandidate);
+            if (customerRequiringApproval.isPresent()) {
+                resourceToImport.setAssociatedArtifacts(convertFilesToPending(resourceToImport));
+                var importedResource = resourceToImport.importResource(publicationService, ImportSource.fromSource(Source.SCOPUS));
+                var userInstanceFromCustomer = createUserInstanceFromCustomer(requestInfo, customerRequiringApproval.get());
+                var fileApproval = PublishingRequestCase.createWithFilesForApproval(
+                    importedResource,
+                    userInstanceFromCustomer,
+                    PublishingWorkflow.lookUp(customerRequiringApproval.get().publicationWorkflow()),
+                    importedResource.getPendingFiles()
+                );
+                fileApproval.persistNewTicket(ticketService);
+                return finalizeImport(importedResource, importCandidate, rawImportCandidate);
+            }
+        }
```

Comment:
This is the main change. When importing ImportCandidate, we persist `associatedCustomers` which are related to contributors that are present on ImportCandidate.

```
When importing ImportCandidate as Publication
And ImportCandidate has files
And first customer we have on ImportCandidate requires approval
Then PendingPublishingRequest is persisted for that customer with pending file to approve. 
```

**And first customer we have on ImportCandidate requires approval** - following condition is unsure for now, cause Frode/ someone else did not decide which customer we want to prioritize, so this may change in future, but main implementation is in place now.